### PR TITLE
Task2 release login

### PR DESCRIPTION
**Pull request by Platton96**

### CODE REVIEW

File: MoneySplitter.Models/Session/LoginModel.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Runtime.Serialization;
+
+namespace MoneySplitter.Models.Session
+{
+    [DataContract]
```
Comment (Vovanella95) [inline MoneySplitter.Models/Session/LoginModel.cs]: Please use different models for Api Revices calls and for using in application

File: MoneySplitter.Services/API/ApiUrlBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace MoneySplitter.Services.API
+{
+    public  class ApiUrlBuilder
+    {
+        private const string WebApiUrl = "http://moneytransfer.azurewebsites.net";
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/ApiUrlBuilder.cs]: Add this to defines class

File: MoneySplitter.Services/API/ApiUrlBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        public Uri Authorization()
+        {
+            return new Uri(string.Concat(
+                WebApiUrl,
+                Defines.Api.Session.SESSION,
+                Defines.Api.Session.SIGN_IN));
+        }
+    }
+
+    public static class Defines
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/ApiUrlBuilder.cs]: Create separate file for this

File: MoneySplitter.Services/API/SessionApiService.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
+﻿using System.Text;
+using System.Net.Http;
+using Newtonsoft.Json;
+using System.Threading.Tasks;
+using MoneySplitter.Models;
+using MoneySplitter.Models.Session;
+using System.Net;
+
+namespace MoneySplitter.Services.API
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/SessionApiService.cs]: MoneySplitter.Services.Api

File: MoneySplitter.Services/MembershipService.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class MembershipService
+    {
+        private readonly SessionApiService _sessionApiServices;
+
+        public UserModel CurrentUser { get; private set; }
+
+        public MembershipService()
+        {
+            _sessionApiServices = new SessionApiService();
```
Comment (Vovanella95) [inline MoneySplitter.Services/MembershipService.cs]: use DI to get services

File: MoneySplitter.Win10/App.xaml.cs [hunk 6]
```diff
@@ -1,100 +1,64 @@
...
+            _container.RegisterNavigationService(rootFrame);
+        }
 
-                if (e.PreviousExecutionState == ApplicationExecutionState.Terminated)
-                {
-                    //TODO: Load state from previously suspended application
-                }
+        protected override void OnLaunched(LaunchActivatedEventArgs args)
+        {
+            if (args.PreviousExecutionState == ApplicationExecutionState.Running)
```
Comment (Vovanella95) [inline MoneySplitter.Win10/App.xaml.cs]: Add { }

File: MoneySplitter.Win10/Common/NavigateManager.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+namespace MoneySplitter.Win10.Common
+{
+    public class NavigationManager : INavigationManager
+    {
+        private readonly INavigationService _navigationService;
+
+        public NavigationManager(INavigationService navigationService)
+        {
+            _navigationService = navigationService;
+            SystemNavigationManager.GetForCurrentView().BackRequested += OnBackRequested; ;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/NavigateManager.cs]: double ;; at the end

File: MoneySplitter.Win10/ViewModels/LoginViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public class LoginViewModel:Screen
+    {
+        private readonly INavigationManager _navigationManager;
+        private MembershipService _membershipService;
+
+        public LoginViewModel( INavigationManager navigationManager)
+        {
+            _navigationManager = navigationManager;
+            _membershipService = new MembershipService();
```
Comment (Vovanella95) [inline MoneySplitter.Win10/ViewModels/LoginViewModel.cs]: use DI container to ger services

File: MoneySplitter.Win10/ViewModels/LoginViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            get { return _email; }
+            set
+            {
+                _email = value;
+                NotifyOfPropertyChange(nameof(Email));
+            }
+        }
+
+        private string _password;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/ViewModels/LoginViewModel.cs]: Please contain fields separatly from Properties

File: MoneySplitter.Infrastructure/INavigationManager.cs [hunk 10]
```diff
@@ -0,0 +1,9 @@
+﻿using MoneySplitter.Models;
+
+namespace MoneySplitter.Infrastructure
+{
+    public interface INavigationManager
+    {
+        void NavigateToShellView(UserModel userModel);
```
Comment (Vovanella95) [inline MoneySplitter.Infrastructure/INavigationManager.cs]: ShellViewModel should get userdata by itself (From membership service). It not need additional parameters

File: MoneySplitter.Models/TransactionModel.cs [hunk 11]
```diff
@@ -1,7 +1,8 @@
 ﻿using System;
-
-namespace MoneySplitter.Models.Entities
+using System.Runtime.Serialization;
+namespace MoneySplitter.Models
 {
+    [DataContract]
```
Comment (Vovanella95) [inline MoneySplitter.Models/TransactionModel.cs]: Models for Api in .Models project again?

File: MoneySplitter.Models/UserModel.cs [hunk 12]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace MoneySplitter.Models
+{
+    [DataContract]
+    public class UserModel
+    {
+        [DataMember]
```
Comment (Vovanella95) [inline MoneySplitter.Models/UserModel.cs]: don't use such models on models project. This is model for API

File: MoneySplitter.Infrastructure/INavigationManager.cs (hunk 10 above)
Comment (Platton96) [reply MoneySplitter.Infrastructure/INavigationManager.cs]: moved LogiModel

File: MoneySplitter.Models/TransactionModel.cs (hunk 11 above)
Comment (Platton96) [reply MoneySplitter.Models/TransactionModel.cs]: Add defines class

File: MoneySplitter.Win10/ViewModels/ShellViewModel.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
+﻿using Caliburn.Micro;
+using MoneySplitter.Models;
+
+namespace MoneySplitter.Win10.ViewModels
+{
+    public class ShellViewModel : Screen
+    {
+        private UserModel _user;
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/ViewModels/ShellViewModel.cs]: rename

File: MoneySplitter.Infrastructure/INavigationManager.cs (hunk 10 above)
Comment (Platton96) [reply MoneySplitter.Infrastructure/INavigationManager.cs]: fixed. removed parametr

**Pull request closed without merging**